modules/data.remote/R/GEDI_AGB_prep.R

```diff
@@ -56,481 +92,437 @@ GEDI_AGB_prep <- function(site_info, time_points, outdir = file.path(getwd(), "G
+      diff.lat <- abs(lat - dat$lat_lowestmode)
+      diff.lon <- abs(lon - dat$lon_lowestmode)
+      keep.inds <- which(diff.lat <= buffer & diff.lon <= buffer)
+      if (length(keep.inds) > 0) {
+        return(1)
       } else {
-        # download GEDI AGB for current site.
-        res.current <- GEDI_AGB_download(start_date = start_date,
-                                         end_date = end_date, 
-                                         outdir = site_folder, 
-                                         extent = extent,
-                                         nfile.min = nfile.min,
-                                         nrow.min = nrow.min,
-                                         gradient = gradient)
-        # if we have previous downloaded GEDI records.
-        if (exists("res", mode = "environment") & !all(is.na(res.current))) {
-          res <- rbind(res, res.current)
-        } else if (!all(is.na(res.current))) {
-          res <- res.current
-        }
-        # write into new csv file.
-        if (exists("res") & !all(is.na(res.current))) {
-          utils::write.csv(res, file = file.path(site_folder, "GEDI_AGB.csv"), row.names = F)
-          # save plot.
-          grDevices::png(file.path(site_folder, "plot.png"))
-          print(GEDI_AGB_plot(outdir = outdir, site.id = point$site_id, start_date = start_date, end_date = end_date))
-          grDevices::dev.off()
-          return(res.current)
-        } else {
-          return(NA)
-        }
+        return(0)
       }
-    }, .progress = T) %>% purrr::set_names(site_info$site_id)
-  GEDI_AGB
+    } %>% unlist
+    # load the file path to the corresponding site if intercepted.
+    for (j in which(res == 1)) {
+      which.point.in.which.file[[j]] <- c(which.point.in.which.file[[j]], files[i])
+    }
+    # update progress bar.
+    utils::setTxtProgressBar(pb, i)
+  }
+  # stop parallel.
+  parallel::stopCluster(cl)
+  foreach::registerDoSEQ()
+  return(which.point.in.which.file)
 }
-#' Download GEDI AGB data for the GEDI AGB extract function.
-#'
-#' @param start_date start date (date with YYYY-MM-DD format) for downloading GEDI AGB from remote database.
-#' @param end_date end date (date with YYYY-MM-DD format) for downloading GEDI AGB from remote database.
-#' @param outdir Directory where the final CSV file will be stored.
-#' @param extent the XY box (in degrees) for downloading GEDI AGB file.
-#' @param nfile.min the minimum required file number to be downloaded and extracted, default is 0.
-#' @param nrow.min the minimum required observation number to be extracted, default is 0.
-#' @param gradient the gradient for iteratively enlarge the extent if the nfile.min or nrow.min are not reached, default is 0. If nfile.min or nrow.min is 0 this will be skipped.
+#' Submit jobs through `qsub` for the `GEDI_L4A_Finder` function.
 #'
-#' @return A data frame containing AGB and sd for the target spatial and temporal extent.
+#' @title GEDI_L4A_Finder_batch
```

Comment:
As above -- "submit jobs through qsub..." is a much better title than the function name is!
```suggestion
```